lib/Provider/MailService.php

```diff
@@ -0,0 +1,202 @@
+<?php
+
+declare(strict_types=1);
+
+/**
+ * SPDX-FileCopyrightText: 2024 Nextcloud GmbH and Nextcloud contributors
+ * SPDX-License-Identifier: AGPL-3.0-or-later
+ */
+namespace OCA\Mail\Provider;
+
+use OCP\Mail\Provider\Address;
+use OCP\Mail\Provider\IAddress;
+use OCP\Mail\Provider\IMessage;
+use OCP\Mail\Provider\IMessageSend;
+use OCP\Mail\Provider\IService;
+use OCP\Mail\Provider\Message;
+
+use Psr\Container\ContainerInterface;
+
+class MailService implements IService, IMessageSend {
+
+	protected array $serviceSecondaryAddresses = [];
+	protected array $serviceAbilities = [];
+
+	public function __construct(
+		protected ContainerInterface $container,
+		protected string $userId = '',
```

Comment:
I'm not a fan of injecting data like userId, except in controllers. this makes the service unusable for background operations, or when you run a procedure for user a in a request authenticated for user b. Maybe you can push the userId into a method parameter.